backend/src/Squidex/Config/Authentication/IdentityServerServices.cs

```diff
@@ -50,15 +51,17 @@ public static AuthenticationBuilder AddSquidexIdentityServerAuthentication(this
                 });
             }
 
-            authBuilder.Services.AddSingleton<IPostConfigureOptions<OpenIdConnectOptions>>(c => new PostConfigureOptions<OpenIdConnectOptions>(OpenIdConnectDefaults.AuthenticationScheme, options =>
+            authBuilder.AddOpenIdConnect(OpenIdConnectDefaults.AuthenticationScheme, options =>
             {
                 if (!string.IsNullOrWhiteSpace(identityOptions.AuthorityUrl))
                 {
                     options.Authority = identityOptions.AuthorityUrl;
                 }
                 else
                 {
-                    options.Authority = GetAuthorityUrl(c);
+                    // probably a better way to acquire DI but this works
+                    var provider = authBuilder.Services.BuildServiceProvider();
+                    options.Authority = GetAuthorityUrl(provider);
```

Comment:
I don't like this, it is really a hack, we should try to understand the options system better.

Reply:
You new code works, and is MUCH nicer. Will rebase,